code/game/machinery/dehumidifier.dm

```diff
@@ -0,0 +1,156 @@
+//This magic dehumidifier is a portable water-to-nothing machine, although it does need a charged cell to work.
+//See drain.dm!
+/obj/machinery/dehumidifier
+	use_power = POWER_USE_OFF
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "dhum-off"
+	name = "emergency dehumidifier"
+	desc = "A portable dehumidifier with emergency high-volume pump to help keep you dry when it's getting damp... just very slowly."
+	var/obj/item/weapon/cell/cell
+	var/on = 0
+	var/drainage = 1
+	var/last_gurgle = 0
+	atom_flags = ATOM_FLAG_CLIMBABLE
+	clicksound = "switch"
+
+/obj/machinery/dehumidifier/Initialize()
+	..()
+	cell = new/obj/item/weapon/cell/high(src)
+	update_icon()
+
+/obj/machinery/dehumidifier/on_update_icon(var/rebuild_overlay = 0)
+	if(!on)
+		icon_state = "dhum-off"
+	else
+		icon_state = "dhum-on"
+
+	if(rebuild_overlay)
+		cut_overlays()
+		if(panel_open)
+			add_overlay("dhum-open")
+
+/obj/machinery/dehumidifier/examine(mob/user)
+	. = ..(user)
+
+	to_chat(user, "The dehumidifier is [on ? "on" : "off"] and the hatch is [panel_open ? "open" : "closed"].")
+	if(panel_open)
+		to_chat(user, "The power cell is [cell ? "installed" : "missing"].")
+	else
+		to_chat(user, "The charge meter reads [cell ? round(cell.percent(),1) : 0]%")
+
+/obj/machinery/dehumidifier/emp_act(severity)
+	if(stat & (BROKEN|NOPOWER))
+		..(severity)
+		return
+	if(cell)
+		cell.emp_act(severity)
+	..(severity)
+
+/obj/machinery/dehumidifier/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/cell))
+		if(panel_open)
+			if(cell)
+				to_chat(user, "There is already a power cell inside.")
+				return
+			else
+				// insert cell
+				if(!user.unEquip(I, src))
+					return
+				cell = I
+				user.visible_message("<span class='notice'>[user] inserts a power cell into [src].</span>", "<span class='notice'>You insert the power cell into [src].</span>")
+				power_change()
+		else
+			to_chat(user, "The hatch must be open to insert a power cell.")
+			return
+	else if(isScrewdriver(I))
+		panel_open = !panel_open
+		user.visible_message("<span class='notice'>[user] [panel_open ? "opens" : "closes"] the hatch on [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the hatch on [src].</span>")
+		update_icon(1)
+		if(!panel_open && user.machine == src)
+			show_browser(user, null, "window=dehumidifier")
+			user.unset_machine()
+	else
+		..()
+
+/obj/machinery/dehumidifier/attack_hand(mob/user as mob)
+	..()
+	interact(user)
+
+/obj/machinery/dehumidifier/interact(mob/user as mob)
```

Comment:
Rather than using Topic and a UI, why not just
````
/obj/machinery/dehumidifier/examine(var/mob/user)
	. = ..()
	if(.)
		to_chat(user, SPAN_NOTICE("The power display reads [cell ? round(cell.percent(),1) : 0]%."))
		if(panel_open)
			if(cell)
				to_chat(user, SPAN_NOTICE("The maintenance panel is open and \the [cell] is installed."))
			else
				to_chat(user, SPAN_WARNING("The maintenance panel is open and there is no cell installed."))

/obj/machinery/dehumidifier/interact(var/mob/user)
	if(Adjacent(user) && panel_open)
		if(cell)
			cell.dropInto(loc)
			user.put_in_hands(cell)
			user.visible_message(SPAN_NOTICE("\The [user] pops \the [cell] out of \the [src]."))
			cell = null
		else
			to_chat(user, SPAN_WARNING("\The [src] has no cell installed."))
	else

		if(!cell || !cell.charge)
			to_chat(user, SPAN_WARNING("You try to switch \the [src] on, but nothing happens."))
			return

		on = !on
		user.visible_message(SPAN_NOTICE("\The [user] switches \the [src] [on ? "on" : "off"]."))
		update_icon()
````

Reply:
If I add proper functionality later it'll want a UI. That said it'll want Nano, so I'll just wipe the UI out completely for now. Makes sense.